src/main/java/net/mcreator/ui/component/CollapsiblePanel.java

```diff
@@ -29,122 +30,60 @@ public class CollapsiblePanel extends JPanel {
 
 	private String title = "";
 	private final TitledBorder border;
+	protected final JPanel contentHolder = new JPanel();
 
 	public CollapsiblePanel(String text, JComponent content) {
-		border = BorderFactory.createTitledBorder(title);
-		setBorder(border);
-		BorderLayout borderLayout = new BorderLayout();
-		setLayout(borderLayout);
+		this(text, content, true);
+	}
+
+	public CollapsiblePanel(String text, JComponent content, boolean allowExpandInitially) {
+		setLayout(new BorderLayout());
+		setOpaque(false);
+
+		setBorder(border = BorderFactory.createTitledBorder(title));
+		setTitle(text);
+		updateBorderTitle();
+
+		contentHolder.setOpaque(false);
+		contentHolder.add(content);
+		super.addImpl(contentHolder, null, -1);
+		toggleVisibility(PreferencesManager.PREFERENCES.ui.expandSectionsByDefault && allowExpandInitially);
+
 		addMouseListener(new MouseAdapter() {
 			@Override public void mouseClicked(MouseEvent e) {
 				toggleVisibility();
 			}
 		});
-		setTitle(text);
-		add(content);
-		updateBorderTitle();
-		setOpaque(false);
 	}
 
-	private final ComponentListener contentComponentListener = new ComponentAdapter() {
-		@Override public void componentShown(ComponentEvent e) {
-			updateBorderTitle();
-		}
-
-		@Override public void componentHidden(ComponentEvent e) {
-			updateBorderTitle();
-		}
-	};
-
 	public void setTitle(String title) {
 		firePropertyChange("title", this.title, this.title = title);
 	}
 
-	@Override public Component add(Component comp) {
-		comp.addComponentListener(contentComponentListener);
-		Component r = super.add(comp);
-		updateBorderTitle();
-		return r;
-	}
-
-	@Override public Component add(String name, Component comp) {
-		comp.addComponentListener(contentComponentListener);
-		Component r = super.add(name, comp);
+	@Override protected void addImpl(Component comp, Object constraints, int index) {
+		contentHolder.add(comp, constraints, index);
 		updateBorderTitle();
-		return r;
-	}
-
-	@Override public Component add(Component comp, int index) {
-		comp.addComponentListener(contentComponentListener);
-		Component r = super.add(comp, index);
-		updateBorderTitle();
-		return r;
-	}
-
-	@Override public void add(Component comp, Object constraints) {
-		comp.addComponentListener(contentComponentListener);
-		super.add(comp, constraints);
-		updateBorderTitle();
-	}
-
-	@Override public void add(Component comp, Object constraints, int index) {
-		comp.addComponentListener(contentComponentListener);
-		super.add(comp, constraints, index);
-		updateBorderTitle();
-	}
-
-	@Override public void remove(int index) {

```

Comment:
It does not seem that you provided an alternative for removal methods

Reply:
These methods were needed because of `contentComponentListener` which is no longer used because its job can be done without its help.

Reply:
If one calls remove(Component), for example, the component will not be removed from the panel as you add the component to the child panel, so it is needed

Reply:
Ah, oops...

Reply:
✔️ Fixed